test/libgit2.jl

```diff
@@ -37,6 +37,15 @@ function with_libgit2_temp_home(f)
     end
 end
 
+function is_defined_remote(repo::LibGit2.GitRepo, remote::AbstractString)
+    try
+        LibGit2.get(LibGit2.GitRemote, repo, name)
```

Comment:
I'll look into this again tomorrow. I didn't find a LibGit2 method which seemed to cover this.